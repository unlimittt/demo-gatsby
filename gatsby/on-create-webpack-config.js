const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
