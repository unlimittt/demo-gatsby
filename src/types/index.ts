import { ReactNode } from "react"
import { FixedObject, FluidObject } from "gatsby-image"

export type RenderCallback = {
  render: (data: any) => ReactNode
}

export type PageContext = {
  tag: string
  category: string
  currentPage: number
  prevPagePath: string
  nextPagePath: string
  hasPrevPage: boolean
  hasNextPage: boolean
}

export type Node = {
  fields: {
    slug: string
    categorySlug?: string
    tagSlugs?: string[]
  }
  frontmatter: {
    title: string
    description?: string
    date?: string
    category?: string
    tags?: string[]
    socialImage?: string
    image?: {
      childImageSharp: {
        fixed?: FixedObject
        fluid?: FluidObject
      }
    }
    order?: number
    price?: string
  }
  html: string
  id: string
}

export type Edge = {
  node: Node
}

export type Edges = Array<Edge>

export type AllMarkdownRemark = {
  allMarkdownRemark: {
    edges: Edges
    group: {
      fieldValue: string
      totalCount: number
    }[]
  }
}

export type MarkdownRemark = Node

export type SiteMetadata = {
  site: {
    siteMetadata: {
      author: {
        name: string
        bio: string
        photo: string
        contacts: {
          facebook: string
          linkedin: string
          github: string
          twitter: string
          telegram: string
          instagram: string
          email: string
          rss: string
          vkontakte: string
          line: string
          gitlab: string
          weibo: string
          codepen: string
          youtube: string
          soundcloud: string
        }
      }
      menu: {
        label: string
        path: string
      }[]
      url: string
      title: string
      subtitle: string
      copyright: string
      disqusShortName: string
    }
  }
}