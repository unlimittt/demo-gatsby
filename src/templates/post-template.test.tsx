import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import PostTemplate from './post-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import markdownRemark from '../../jest/__fixtures__/markdown-remark';
import { RenderCallback } from '../types';
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe('PostTemplate', () => {
  const props = {
    data: {
      ...markdownRemark
    }
  };

  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const wrapper = shallow(<PostTemplate {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
