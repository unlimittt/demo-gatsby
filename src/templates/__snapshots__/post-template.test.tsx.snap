// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders correctly 1`] = `
<Layout
  description="test"
  title="test"
>
  <Post
    post={
      Object {
        "fields": Object {
          "slug": "/test_0",
          "tagSlugs": Array [
            "/test_0",
            "/test_1",
          ],
        },
        "frontmatter": Object {
          "date": "2016-09-01",
          "description": "test",
          "image": Object {
            "childImageSharp": Object {
              "fixed": Object {
                "height": 184,
                "src": "/media/customized-coke-bottle.png",
                "srcSet": "",
                "width": 200,
              },
              "fluid": Object {
                "aspectRatio": 1,
                "sizes": "",
                "src": "/media/customized-coke-bottle.png",
                "srcSet": "",
              },
            },
          },
          "tags": Array [
            "test_0",
            "test_1",
          ],
          "title": "test",
        },
        "html": "<p>test</p>",
        "id": "test-123",
      }
    }
  />
</Layout>
`;
